hw/xbox/xid-sdl.c

```diff
@@ -464,6 +478,63 @@ static void usb_xid_class_initfn(ObjectClass *klass, void *data)
     uc->handle_attach  = usb_desc_attach;
 }
 
+static HapticEmulationMode pick_haptic_emulation_mode(SDL_Haptic *haptic)
+{
+    if (SDL_HapticQuery(haptic) & SDL_HAPTIC_LEFTRIGHT) {
+        return EMU_HAPTIC_LEFT_RIGHT;
+    }
+
+    return EMU_UNSUPPORTED;
+}
+
+static int haptic_init(USBXIDState *s)
+{
+    if (SDL_HapticOpened(s->device_index)) {
+        fprintf(stderr, "Joystick haptic effect already in use, haptic effect disabled\n");
```

Comment:
This isn't about the "haptic effect" being "in use", this is about the "haptic device" being "already opened".